src/Exception/HostedDomainException.php

```diff
@@ -7,8 +7,7 @@
  */
 class HostedDomainException extends \Exception
 {
-
-    public static function notMatchingDomain($configuredDomain)
+    public static function notMatchingDomain($configuredDomain): HostedDomainException
```

Comment:
Static constructors should be like:

```suggestion
    /**
     * @return static
     */
    public static function notMatchingDomain($configuredDomain): self
```